src/Widgets/BridgeWidgetApi.ts

```diff
@@ -1,69 +1,211 @@
     }
 
-    private async getRoomFromRequest(req: Request): Promise<AdminRoom|{error: string, statusCode: number}> {
-        const { roomId } = req.params;
-        const token = req.headers.authorization?.substr('Bearer '.length);
-        if (!token) {
-            return {
-                error: 'Access token not given',
-                statusCode: 400,
-            };
+    private async getRoomFromRequest(req: ProvisioningRequest): Promise<AdminRoom> {
+        const room = [...this.adminRooms.values()].find(r => r.userId === req.userId);
+        if (!room) {
+            throw new ApiError("No room found for access token", ErrCode.BadToken);
         }
-        // Replace with actual auth
-        const room = this.adminRooms.get(roomId);
-        if (!room || !room.verifyWidgetAccessToken(token)) {
-            return {error: 'Unauthorized access to room', statusCode: 401}
+        return room;
+    }
+
+    private async getRoomState(req: ProvisioningRequest, res: Response) {
+        try {
+            const room = await this.getRoomFromRequest(req);
+            await res.send(await room.getBridgeState());
+        } catch (ex) {
+            log.error(`Failed to get room state:`, ex);
+            throw new ApiError("An error occured when getting room state", ErrCode.Unknown);
         }
+    }
 
-        return room;
+    private async getConfigSections(req: ProvisioningRequest, res: Response<{[section: string]: boolean}>) {
+        res.send({
+            general: true,
+            github: !!this.config.github,
+            gitlab: !!this.config.gitlab,
+            generic: !!this.config.generic,
+            jira: !!this.config.jira,
+            figma: !!this.config.figma,
+        });
+    }
+
+    private async getServiceConfig(req: ProvisioningRequest, res: Response<Record<string, unknown>>) {
+        let config: undefined|Record<string, unknown>;
+        switch (req.params.service) {
+            case "generic":
+                config = this.config.generic?.publicConfig;
+                break;
+            default:
+                throw new ApiError("Not a known service, or service doesn't expose a config", ErrCode.NotFound);
+        }
+
+        if (!config) {
+            throw new ApiError("Service is not enabled", ErrCode.DisabledFeature);
+        }
+
+        res.send({
+            config
+        });
+    }
+
+    private async getConfigSection(req: ProvisioningRequest, res: Response<WidgetConfigurationSection[]>) {
+        if (req.params.section !== "general") {
+            throw new ApiError("Not a known config section", ErrCode.NotFound);
+        }
+        res.send(
+            [{
+
+                name: 'Overview',
+                options: [{
+                    key: 'name',
+                    type: WidgetConfigurationType.String,
+                    currentValue: null,
+                    defaultValue: 'Agent Smith',
```

Comment:
...is this a placeholder? :)

Reply:
Oh, this is a horrible placeholder from the admin configuration stuff that I never finished. Hmm, what to do with this...